nifi-nar-bundles/nifi-framework-bundle/nifi-stateless-bundle/nifi-stateless-api/src/main/java/org/apache/nifi/stateless/engine/StatelessEngineConfiguration.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.stateless.engine;
+
+import org.apache.nifi.stateless.config.ExtensionClientDefinition;
+import org.apache.nifi.stateless.config.SslContextDefinition;
+
+import java.io.File;
+import java.util.List;
+
+public interface StatelessEngineConfiguration {
+    File getWorkingDirectory();
+
+    File getNarDirectory();
+
+    File getKrb5File();
+
+    SslContextDefinition getSslContext();
```

Comment:
Yeah, I can buy the argument for calling it getSslContextDefintiion(). We cannot use StandardTlsConfiguration because of the point at which it's being referenced. In order to ensure that extensions run the same way in Stateless NiFi as they do in Traditional NiFi, we have to ensure that the root class path is the same in both cases. Since the module that defines that is not on the root classpath, it cannot be brought in here.